### PR TITLE
fix(NcReferenceWidget): pass elements to intersection observer

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Regression from https://github.com/nextcloud-libraries/nextcloud-vue/pull/5561

Intersection observer has observed elements as output

### 🏁 Checklist

- [x] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/components/NcRichText/NcReferenceWidget.vue [hunk 1]
```diff
@@ -72,9 +72,11 @@ export default {
 		const widgetRoot = ref()
 		const { width } = useElementSize(widgetRoot)
 
-		useIntersectionObserver(widgetRoot, () => {
-			nextTick(() => {
-				isVisible.value = widgetRoot.value?.isIntersecting ?? false
+		useIntersectionObserver(widgetRoot, (entries) => {
```
Comment (Antreesy) [inline src/components/NcRichText/NcReferenceWidget.vue:75]: There is always only one element expected, so we could omit forEach cycle
```suggestion
		useIntersectionObserver(widgetRoot, ([entry]) => {
```

**Review by Antreesy — APPROVED** [review]
Tested

[Screencast from 02.07.2024 15:37:38.webm](https://github.com/nextcloud-libraries/nextcloud-vue/assets/93392545/b6ec59f4-2e30-473e-90f7-c80ed190a9c3)

Shall we keep `?? false` as a fallback?

**Review by susnux — APPROVED** [review]

File: src/components/NcRichText/NcReferenceWidget.vue [hunk 2]
```diff
@@ -72,9 +72,9 @@ export default {
 		const widgetRoot = ref()
 		const { width } = useElementSize(widgetRoot)
 
-		useIntersectionObserver(widgetRoot, () => {
+		useIntersectionObserver(widgetRoot, ([entry]) => {
 			nextTick(() => {
-				isVisible.value = widgetRoot.value?.isIntersecting ?? false
+				isVisible.value = entry.isIntersecting
```
Comment (susnux) [inline src/components/NcRichText/NcReferenceWidget.vue:77]: @Antreesy  I think it is guaranteed to be defined, at least the docs say so

**Comment by susnux** [comment]
Please always add "bug" or "enhancement" labels if the PR is not a refactoring / chore / ci / test only PR

**Pull request merged**